presto-main/src/main/java/com/facebook/presto/memory/NodeMemoryConfig.java

```diff
@@ -25,13 +25,15 @@
 public class NodeMemoryConfig
 {
     public static final long AVAILABLE_HEAP_MEMORY = Runtime.getRuntime().maxMemory();
+    public static final String QUERY_MAX_BROADCAST_MEMORY_CONFIG = "query.max-broadcast-memory";
     public static final String QUERY_MAX_MEMORY_PER_NODE_CONFIG = "query.max-memory-per-node";
     public static final String QUERY_SOFT_MAX_MEMORY_PER_NODE_CONFIG = "query.soft-max-memory-per-node";
     public static final String QUERY_MAX_TOTAL_MEMORY_PER_NODE_CONFIG = "query.max-total-memory-per-node";
     public static final String QUERY_SOFT_MAX_TOTAL_MEMORY_PER_NODE_CONFIG = "query.soft-max-total-memory-per-node";
 
     private boolean isReservedPoolEnabled = true;
 
+    private DataSize maxQueryBroadcastMemory = new DataSize(AVAILABLE_HEAP_MEMORY * 0.1, BYTE);
```

Comment:
This default when nothing is specified won't match maxQueryMemoryPerNode if that is set to some other value and principle of least surprise wise it seems like you would expect them to be the same.
@rschlussel  WDYT?

Reply:
My concern about that was that if you change the session property of maxUserMemory, it would be tricky to know if you should change the value for the maxBroadcastLimit session property.  However, thinking about this more, maxQueryMemoryPerNode is a hard limit, so it would be okay to rely on the configuration property value for setting the default broadcast configuration and leaving the session properties as it currently is (since the main thing we want to avoid is for this value to be lower if the user didn't set it lower).

Reply:
Are you suggesting we always cap the maximum possible value of maxQueryBroadcastMemory with maxQueryMemoryPerNode (e.g., when setMaxQueryMemoryPerNode or setMaxQueryBroadcastMemory is called add an additional checking between two values and set maxBroadcast to maxQueryMemPerNode when it is larger)?

Reply:
i think the suggestion is to set it similar to the way that the softMemoryLimit is set. SoftMaxQueryTotalMemoryPerNode is unset by default, and for getSoftMaxQueryTotalMemoryPerNode, when the value is null, we return the maxMemoryPerNode.